scripts/ci-e2e.sh

```diff
@@ -75,10 +76,56 @@ source "${REPO_ROOT}/hack/ensure-kubectl.sh"
 # shellcheck source=./hack/ensure-kustomize.sh
 source "${REPO_ROOT}/hack/ensure-kustomize.sh"
 
-# This will run the tests with env variabls defined in environment.sh
-# or exported by metal3-dev-env scripts
-${M3_DEV_ENV_PATH}/scripts/run_command.sh make e2e-tests
+# Exported to the cluster templates
+SSH_PUB_KEY_CONTENT=$(file < ${HOME}/.ssh/id_rsa.pub)
+export SSH_PUB_KEY_CONTENT
 
-pushd ${M3_DEV_ENV_PATH} || exit 1
+os_check
+if [[ "${OS}" == ubuntu ]]; then
+    export IMAGE_OS=${IMAGE_OS:-"ubuntu"}
+else
+    export IMAGE_OS=${IMAGE_OS:-"centos"}
+fi
+# TODO FIXME (Mohammed) needed only in the next exports 
+export FROM_K8S_VERSION="v1.23.2"
+export KUBERNETES_VERSION=${FROM_K8S_VERSION}
+export UPGRADED_K8S_VERSION="v1.23.3"
+# The host that has images for provisioning, this should be in the
+# format of a URL host, e.g. with IPv6, it should be surrounded
+# by brackets
+export PROVISIONING_URL_HOST="172.22.0.1"
+# TODO (Mohammed) this was copied from devenv/lib/images.sh can reuse
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
+  export IMAGE_NAME=${IMAGE_NAME:-UBUNTU_20.04_NODE_IMAGE_K8S_${KUBERNETES_VERSION}.qcow2}
+  export IMAGE_LOCATION=${IMAGE_LOCATION:-https://artifactory.nordix.org/artifactory/metal3/images/k8s_${KUBERNETES_VERSION}}
+elif [[ "${IMAGE_OS}" == "centos" ]]; then
+  export IMAGE_NAME=${IMAGE_NAME:-CENTOS_8_NODE_IMAGE_K8S_${KUBERNETES_VERSION}.qcow2}
+  export IMAGE_LOCATION=${IMAGE_LOCATION:-https://artifactory.nordix.org/artifactory/metal3/images/k8s_${KUBERNETES_VERSION}}
+fi
+
+# Image url and checksum
+export IMAGE_URL=http://$PROVISIONING_URL_HOST/images/${IMAGE_NAME}
+export IMAGE_CHECKSUM=http://$PROVISIONING_URL_HOST/images/${IMAGE_NAME}.md5sum
+
+# Image basename and rawname
+IMAGE_BASE_NAME="${IMAGE_NAME%.*}"
+export IMAGE_RAW_NAME="${IMAGE_BASE_NAME}-raw.img"
+
+# variables used for template parametrization in CAPM3
+export IMAGE_RAW_URL=http://$PROVISIONING_URL_HOST/images/${IMAGE_RAW_NAME}
+export IMAGE_RAW_CHECKSUM=http://$PROVISIONING_URL_HOST/images/${IMAGE_RAW_NAME}.md5sum
+
+# UPGRADE
+if [[ "${IMAGE_OS}" == "ubuntu" ]]; then
+  export UPGRADED_IMAGE_NAME="UBUNTU_20.04_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}.qcow2"
+  export UPGRADED_RAW_IMAGE_NAME="UBUNTU_20.04_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}-raw.img"
+else
+  export UPGRADED_IMAGE_NAME="CENTOS_8_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}.qcow2"
+  export UPGRADED_RAW_IMAGE_NAME="CENTOS_8_NODE_IMAGE_K8S_${UPGRADED_K8S_VERSION}-raw.img"
+fi
```

Comment:
Building the image vars based on different os is process implemented in devenv Should not we have a better way to reuse that instead of duplicate it here https://github.com/metal3-io/metal3-dev-env/blob/master/lib/images.sh